editor/src/clj/editor/code/view.clj

```diff
@@ -3354,29 +3355,24 @@
     (when is-code-resource-type
       (g/with-auto-evaluation-context evaluation-context
         (r/ensure-loaded! resource-node evaluation-context)))
-    (let [[lines document-width]
-          (g/with-auto-evaluation-context evaluation-context
-            (let [glyph-metrics (g/node-value view-node :glyph-metrics evaluation-context)
-                  tab-spaces (g/node-value view-node :tab-spaces evaluation-context)
-                  tab-stops (data/tab-stops glyph-metrics tab-spaces)
-                  lines (g/node-value resource-node :lines evaluation-context)
-                  document-width (data/max-line-width glyph-metrics tab-stops lines)]
-              [lines document-width]))]
-      (g/transact
-        (concat
-          (g/set-property view-node :document-width document-width)
-          (g/connect app-view :debugger-execution-locations view-node :debugger-execution-locations)
-          (gu/connect-existing-outputs resource-node-type resource-node view-node
-            [[:completions :completions]
-             [:cursor-ranges :cursor-ranges]
-             [:indent-type :indent-type]
-             [:invalidated-rows :invalidated-rows]
-             [:lines :lines]
-             [:regions :regions]])))
-      (when (and is-code-resource-type
-                 (resource/file-resource? resource))
-        (lsp/open-view! lsp view-node resource lines))
-      view-node)))
+    (g/transact
+      (concat
+        (g/connect app-view :debugger-execution-locations view-node :debugger-execution-locations)
+        (gu/connect-existing-outputs resource-node-type resource-node view-node
+          [[:completions :completions]
+           [:cursor-ranges :cursor-ranges]
+           [:indent-type :indent-type]
+           [:invalidated-rows :invalidated-rows]
+           [:lines :lines]
+           [:regions :regions]])))
+    (g/transact
+      (g/with-auto-evaluation-context evaluation-context
+        (update-document-width evaluation-context view-node)))
+    (when (and is-code-resource-type
+               (resource/file-resource? resource))
+      (let [lines (g/node-value view-node :lines)]
+        (lsp/open-view! lsp view-node resource lines)))
+    view-node))
```

Comment:
Mostly indentation, but the change is to update the `document-width` property of the `view-node` in a separate transaction after it has been connected to the `resource-node`. We need this for the `tab-spaces` output of the `view-node` to be valid. Previously we would evaluate the `tab-spaces` output on the `view-node` before the connection to the `resource-node` had been established, which would cause us to fall back to the default indentation type instead of basing it on the content of the file.